src/ui/view/dataverse/component/pageTemplate/pageTemplate.tsx

```diff
@@ -3,20 +3,28 @@ import { pipe } from 'fp-ts/lib/function'
 import * as A from 'fp-ts/Array'
 import type { DataverseItemDetails } from '@/ui/page/dataverse/dataverse'
 import { Tags } from '@/ui/view/dataverse/component/tags/tags'
-import type { Metadata, TagsMetadata, GeneralMetadata } from '@/ui/view/dataverse/types'
+import type { ItemGeneralMetadata } from '@/ui/view/dataverse/types'
 import { GeneralMetadataList } from '@/ui/view/dataverse/component/generalMetadata/generalMetadata'
 import './pageTemplate.scss'
+import type { IconName } from '@/ui/component/icon/icon'
 
 type PageTemplateProps = {
   data: DataverseItemDetails
-  metadata: Metadata[]
+  metadata: ItemGeneralMetadata[]
 }
 
-const isTagsMetadata = (metadata: Metadata): metadata is TagsMetadata =>
+const isTagsMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is Omit<ItemGeneralMetadata, 'value'> & { property: 'tags'; value: string[] } =>
   metadata.property === 'tags'
 
-const isGeneralMetadata = (metadata: Metadata): metadata is GeneralMetadata =>
-  metadata.property !== 'tags'
+const isGeneralMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is Omit<ItemGeneralMetadata, 'value'> & {
+  property: string
+  value: string
+  iconName?: IconName
+} => metadata.property !== 'tags'
```

Comment:
```suggestion
const isGeneralMetadata = (
  metadata: ItemGeneralMetadata
): metadata is Omit<ItemGeneralMetadata, 'value'> & {
  property: 'tags'
  value: string
} => metadata.property !== 'tags'
```

Reply:
@ccamel  Is it be possible to write it like this: 
```
const isTagsMetadata = (
  metadata: ItemGeneralMetadata
): metadata is Pick<ItemGeneralMetadata, 'property'> & {
  value: string[]
} => metadata.property === 'tags'
```
 and
 
 ```
 const isGeneralMetadata = (
  metadata: ItemGeneralMetadata
): metadata is Omit<ItemGeneralMetadata, 'value'> & {
  value: string
} => metadata.property !== 'tags'
```

Reply:
It depends on the desired scalability. For me, we might want to get all the properties of the original type, but with the property and value narrowed appropriately, hence the `Omit<>` instead of the `Pick`.

Reply:
Very good point 👍